modules/preview-react/side-panel/lib/SidePanel.tsx

```diff
@@ -98,172 +94,126 @@ const containerVariantStyle: Record<SidePanelVariant, CSSObject> = {
+            // mounted.current will be false on the first render, thus you won't get an unwanted animation here
+            // Will animate again if you force a re-render (like in Storybook)
+            animation: touched
+              ? `${expanded ? motion.expand : motion.collapse} 200ms ease-out`
+              : undefined,
+          },
+          containerVariantStyle[variant],
+        ]}
+        onAnimationEnd={handleAnimationEnd}
+        onAnimationStart={handleAnimationStart}
+        {...elemProps}
       >
-        {children}
-      </SidePanelContext.Provider>
-    </Panel>
-  );
-};
-
-export type ToggleButtonProps = TertiaryButtonProps & {
-  /**
-   * The tooltip text to expand the side panel
-   * @default 'Expand'
-   */
-  tooltipTextExpand?: string;
-  /**
-   * The tooltip text to collapse the side panel
-   * @default 'Collapse'
-   */
-  tooltipTextCollapse?: string;
-};
-
-/**
- * A toggle button styled specifically for the side panel container.
- */
-const ToggleButton = ({
-  variant = undefined,
-  icon = transformationImportIcon,
-  tooltipTextExpand: expandLabel = 'Expand',
-  tooltipTextCollapse: collapseLabel = 'Collapse',
-  ...rest
-}: ToggleButtonProps & React.ButtonHTMLAttributes<HTMLButtonElement>) => {
-  const context = React.useContext(SidePanelContext);
-
-  const useRTLOrigin = () => {
-    const isRTL = useIsRTL();
-    // if the direction is set to RTl, flip the origin
-    if (isRTL) {
-      return context.origin === 'left' ? 'right' : 'left';
-    }
-    // Otherwise, default to returning the origin
-    return context.origin;
-  };
-
-  const rtlOrigin = useRTLOrigin();
-
-  // Note: Depending on the collapsed width, the button could "jump" to it's final position.
-  const buttonStyle = css({
-    position: 'absolute',
-    top: space.m,
-    width: space.l,
-    right: context.state === 'collapsed' ? 0 : rtlOrigin === 'left' ? space.s : undefined,
-    left: context.state === 'collapsed' ? 0 : rtlOrigin === 'right' ? space.s : undefined,
-    margin: context.state === 'collapsed' ? 'auto' : 0, // to override the -8px margin for TertiaryButton.Plain
-    transform:
-      context.state === 'collapsed' || context.state === 'collapsing'
-        ? `scaleX(${rtlOrigin === 'left' ? '1' : '-1'})`
-        : `scaleX(${rtlOrigin === 'left' ? '-1' : '1'})`,
-  });
-
-  return (
-    <Tooltip title={context.state === 'collapsed' ? expandLabel : collapseLabel} type="muted">
-      <TertiaryButton type="button" css={buttonStyle} icon={icon} variant={variant} {...rest} />
-    </Tooltip>
-  );
-};
-
-SidePanel.ToggleButton = ToggleButton;
+        <SidePanelContext.Provider
+          value={{
+            state,
+            origin,
+          }}
+        >
+          {children}
+        </SidePanelContext.Provider>
+      </Panel>
+    );
+  },
+  subComponents: {
+    /**
+     * `<SidePanel.ToggleButton>` is a control that is meant to toggle between `expanded = true` and
```

Comment:
Let's remove the angle brackets here since we don't typically use them in our documentation.
```suggestion
     * `SidePanel.ToggleButton` is a control that is meant to toggle between `expanded = true` and
```